src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Embeds the App Name into the Nuetered host  
+    /// </summary>
+    public class EmbedAppNameInHost : TaskBase
+    {
+        private ITaskItem _outputAppHostPath;
+
+        [Required]
+        public string AppHostSourcePath { get; set; }
+
+        [Required]
+        public string AppHostDestinationDirectoryPath { get; set; }
+
+        [Required]
+        public string AppName { get; set; }
+
+        [Required]
+        public string StringToReplace { get; set; }
+
+        [Output]
+        public ITaskItem ModifiedAppHostPath
+        {
+            get { return _outputAppHostPath; }
+        }
+
+        protected override void ExecuteCore()
+        {
+            var array = File.ReadAllBytes(AppHostSourcePath);
```

Comment:
This is a one time operation for every project. We have made the design decision to replace the byte stream in the host, i suppose there is no way out

On win7-x64 it is 74 kb